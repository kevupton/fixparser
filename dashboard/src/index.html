<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, minimum-scale=1,initial-scale=1" />
    <meta name="description" content="FIX Parser, ECMAScript framework for working with FIX protocol messages." />
    <link href="dashboard.css" rel="stylesheet">
    <title>FIX Parser</title>
</head>
<body>
<div id="application"></div>
</body>
</html>
