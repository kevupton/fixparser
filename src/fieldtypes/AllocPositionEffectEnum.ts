/*
 * fixparser
 * https://gitlab.com/logotype/fixparser.git
 *
 * Copyright Â© 2022 fixparser.io
 * Released under Commercial license. Check LICENSE.md
 */
export enum AllocPositionEffectEnum {
    Open = 'O',
    Close = 'C',
    Rolled = 'R',
    Fifo = 'F',
}
