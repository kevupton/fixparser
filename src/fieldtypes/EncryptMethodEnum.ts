/*
 * fixparser
 * https://gitlab.com/logotype/fixparser.git
 *
 * Copyright Â© 2022 fixparser.io
 * Released under Commercial license. Check LICENSE.md
 */
export enum EncryptMethodEnum {
    None = 0,
    PKCS = 1,
    DES = 2,
    PKCSDES = 3,
    PGPDES = 4,
    PGPDESMD5 = 5,
    PEM = 6,
}
