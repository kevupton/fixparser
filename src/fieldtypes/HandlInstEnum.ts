/*
 * fixparser
 * https://gitlab.com/logotype/fixparser.git
 *
 * Copyright Â© 2022 fixparser.io
 * Released under Commercial license. Check LICENSE.md
 */
export enum HandlInstEnum {
    AutomatedExecutionNoIntervention = 1,
    AutomatedExecutionInterventionOK = 2,
    ManualOrder = 3,
}
