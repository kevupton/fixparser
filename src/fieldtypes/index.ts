/*
 * fixparser
 * https://gitlab.com/logotype/fixparser.git
 *
 * Copyright Â© 2022 fixparser.io
 * Released under Commercial license. Check LICENSE.md
 */
export { AllocPositionEffectEnum as AllocPositionEffect } from './AllocPositionEffectEnum';
export { EncryptMethodEnum as EncryptMethod } from './EncryptMethodEnum';
export { ExecTypeEnum as ExecType } from './ExecTypeEnum';
export { FieldEnum as Fields } from './FieldEnum';
export { HandlInstEnum as HandlInst } from './HandlInstEnum';
export { MarketDepthEnum as MarketDepth } from './MarketDepthEnum';
export { MDEntryTypeEnum as MDEntryType } from './MDEntryTypeEnum';
export { MDUpdateTypeEnum as MDUpdateType } from './MDUpdateTypeEnum';
export { MessageEnum as Messages } from './MessageEnum';
export { OrderStatusEnum as OrderStatus } from './OrderStatusEnum';
export { OrderTypesEnum as OrderTypes } from './OrderTypesEnum';
export { SideEnum as Side } from './SideEnum';
export { SubscriptionRequestTypeEnum as SubscriptionRequestType } from './SubscriptionRequestTypeEnum';
export { TimeInForceEnum as TimeInForce } from './TimeInForceEnum';
