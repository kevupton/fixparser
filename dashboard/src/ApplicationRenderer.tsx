import { render } from 'react-dom';
import * as UIkit from 'uikit';
import * as Icons from 'uikit/dist/js/uikit-icons';

import './dashboard.scss';
import { Dashboard } from './Dashboard';

UIkit.use(Icons);

const container: HTMLElement = document.createElement('application');
container.className = 'application flexContainer flexColumn';
document.body.appendChild(container);

render(<Dashboard />, container);
