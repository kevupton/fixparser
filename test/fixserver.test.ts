import { FIXServer, EncryptMethod, Field, Fields, Message, Messages, Protocol } from '../src/FIXServer';
import { FIXParser, HandlInst, OrderTypes, Side, TimeInForce } from '../src/FIXParser';
import { mockLicense } from './setup';
import { FieldEnum } from '../src/fieldtypes/FieldEnum';

jest.setTimeout(30000);

describe('FIXServer', () => {
    let fixServer: FIXServer = new FIXServer();
    let fixParser: FIXParser = new FIXParser();
    let port: number = 9800;

    describe('FIXServer', () => {
        it('#connect with no license', () => {
            mockLicense.mockReturnValue(false);
            fixServer = new FIXServer();
            fixParser = new FIXParser();
            // Connect with a client
            fixParser.connect({
                host: 'localhost',
                port,
                protocol: 'tcp',
            });
            expect(mockLicense).toHaveBeenCalled();
            mockLicense.mockReturnValue(true);
        });
    });

    describe('TCP', () => {
        const PROTOCOL: Protocol = 'tcp';

        it('End-to-end: connect and Logon', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER',
                target: 'CLIENT',
                logging: false,
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'CLIENT',
                target: 'SERVER',
                fixVersion: 'FIX.4.7',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send a Logon message
                const logon: Message = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.Logon),
                    new Field(Fields.MsgSeqNum, fixParser.getNextTargetMsgSeqNum()),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ResetSeqNumFlag, 'Y'),
                    new Field(Fields.EncryptMethod, EncryptMethod.None),
                    new Field(Fields.HeartBtInt, 64),
                );
                fixParser.send(logon);
            });
            fixParser.on('message', (message: Message) => {
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
                expect(fixServer.heartBeatInterval).toEqual(64);
                expect(fixServer.fixVersion).toEqual('FIX.4.7');
                fixParser.close();
                fixServer.destroy();
                done();
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });

        it('End-to-end: invalid Logon', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER',
                target: 'CLIENT',
                logging: false,
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'INVALID_SENDER',
                target: 'INVALID_TARGET',
                fixVersion: 'FIX.5.0',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send a Logon message
                const logon: Message = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.Logon),
                    new Field(Fields.MsgSeqNum, fixParser.getNextTargetMsgSeqNum()),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ResetSeqNumFlag, 'Y'),
                    new Field(Fields.EncryptMethod, EncryptMethod.None),
                    new Field(Fields.HeartBtInt, 60),
                );
                fixParser.send(logon);
            });
            fixParser.on('message', (message: Message) => {
                expect(message.description).toEqual('Logout');
                expect(message.messageString).toMatchSnapshot();
                fixParser.close();
                fixServer.destroy();
                done();
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });

        it('End-to-end: invalid MsgSeqNum', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER',
                target: 'CLIENT',
                logging: false,
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'CLIENT',
                target: 'SERVER',
                fixVersion: 'FIX.4.7',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send a Logon message
                const logon: Message = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.Logon),
                    new Field(Fields.MsgSeqNum, 100),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ResetSeqNumFlag, 'Y'),
                    new Field(Fields.EncryptMethod, EncryptMethod.None),
                    new Field(Fields.HeartBtInt, 64),
                );
                fixParser.send(logon);
            });
            fixParser.on('message', (message: Message) => {
                expect(message.description).toEqual('ResendRequest');
                expect(message.messageString).toMatchSnapshot();
                expect(fixServer.heartBeatInterval).toEqual(30);
                expect(fixServer.fixVersion).toEqual('FIX.4.7');
                fixParser.close();
                fixServer.destroy();
                done();
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });

        it('End-to-end: first message not a Logon', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER',
                target: 'CLIENT',
                logging: false,
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('NewOrderSingle');
                expect(message.messageString).toMatchSnapshot();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'CLIENT',
                target: 'SERVER',
                fixVersion: 'FIX.4.8',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send anything but a Logon
                const wrongFirstMessage = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.NewOrderSingle),
                    new Field(Fields.MsgSeqNum, fixParser.getNextTargetMsgSeqNum()),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ClOrdID, '11223344'),
                    new Field(Fields.HandlInst, HandlInst.AutomatedExecutionNoIntervention),
                    new Field(Fields.OrderQty, '123'),
                    new Field(Fields.TransactTime, fixParser.getTimestamp()),
                    new Field(Fields.OrdType, OrderTypes.Market),
                    new Field(Fields.Side, Side.Buy),
                    new Field(Fields.Symbol, '700.HK'),
                    new Field(Fields.TimeInForce, TimeInForce.Day),
                );
                fixParser.send(wrongFirstMessage);
            });
            fixParser.on('message', (message: Message) => {
                expect(message.description).toEqual('Logout');
                expect(message.messageString).toMatchSnapshot();
                expect(message.getField(FieldEnum.Text)!.value).toEqual('First message not a Logon');
                expect(fixServer.heartBeatInterval).toEqual(30);
                expect(fixServer.fixVersion).toEqual('FIX.5.0SP2');
                fixParser.close();
                fixServer.destroy();
                done();
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });
    });

    describe('Websocket', () => {
        const PROTOCOL: Protocol = 'websocket';

        it('End-to-end: connect and Logon', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER2',
                target: 'CLIENT2',
                logging: false,
            });
            fixServer.on('error', (error) => {
                console.log('FIXServer: ', error);
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
                expect(fixServer.heartBeatInterval).toEqual(128);
                expect(fixServer.fixVersion).toEqual('FIX.4.6');
                fixParser.close();
                fixServer.destroy();
                done();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'CLIENT2',
                target: 'SERVER2',
                fixVersion: 'FIX.4.6',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send a Logon message
                const logon: Message = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.Logon),
                    new Field(Fields.MsgSeqNum, fixParser.getNextTargetMsgSeqNum()),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ResetSeqNumFlag, 'Y'),
                    new Field(Fields.EncryptMethod, EncryptMethod.None),
                    new Field(Fields.HeartBtInt, 128),
                );
                fixParser.send(logon);
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });

        it('End-to-end: invalid Logon', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER33',
                target: 'CLIENT33',
                logging: false,
            });
            fixServer.on('error', (error) => {
                console.log('FIXServer: ', error);
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'INVALID_SENDER',
                target: 'INVALID_TARGET',
                fixVersion: 'FIX.5.0',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send a Logon message
                const logon: Message = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.Logon),
                    new Field(Fields.MsgSeqNum, fixParser.getNextTargetMsgSeqNum()),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ResetSeqNumFlag, 'Y'),
                    new Field(Fields.EncryptMethod, EncryptMethod.None),
                    new Field(Fields.HeartBtInt, 60),
                );
                fixParser.send(logon);
            });
            fixParser.on('message', (message: Message) => {
                expect(message.description).toEqual('Logout');
                expect(message.messageString).toMatchSnapshot();
                fixParser.close();
                fixServer.destroy();
                done();
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });

        it('End-to-end: invalid MsgSeqNum', (done) => {
            port += 2;

            fixServer = new FIXServer();
            fixParser = new FIXParser();
            const HOST: string = 'localhost';

            // Start up a server
            fixServer.createServer({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'SERVER2',
                target: 'CLIENT2',
                logging: false,
            });
            fixServer.on('error', (error) => {
                console.log('FIXServer: ', error);
            });

            // Listen for messages
            fixServer.on('message', (message: Message) => {
                expect(fixServer.isConnected()).toBeTruthy();
                expect(fixParser.isConnected()).toBeTruthy();
                expect(message.description).toEqual('Logon');
                expect(message.messageString).toMatchSnapshot();
                expect(fixServer.heartBeatInterval).toEqual(30);
                expect(fixServer.fixVersion).toEqual('FIX.4.6');
                fixParser.close();
                fixServer.destroy();
                done();
            });

            expect(fixServer.isConnected()).toBeFalsy();
            expect(fixParser.isConnected()).toBeFalsy();

            // Connect with a client
            fixParser.connect({
                host: HOST,
                port,
                protocol: PROTOCOL,
                sender: 'CLIENT2',
                target: 'SERVER2',
                fixVersion: 'FIX.4.6',
                logging: false,
            });

            expect(mockLicense).toHaveBeenCalled();

            fixParser.on('open', () => {
                expect(fixParser.isConnected()).toBeTruthy();

                // Send a Logon message
                const logon: Message = fixParser.createMessage(
                    new Field(Fields.MsgType, Messages.Logon),
                    new Field(Fields.MsgSeqNum, 200),
                    new Field(Fields.SenderCompID, fixParser.sender),
                    new Field(Fields.SendingTime, fixParser.getTimestamp()),
                    new Field(Fields.TargetCompID, fixParser.target),
                    new Field(Fields.ResetSeqNumFlag, 'Y'),
                    new Field(Fields.EncryptMethod, EncryptMethod.None),
                    new Field(Fields.HeartBtInt, 128),
                );
                fixParser.send(logon);
            });
            fixParser.on('message', (message: Message) => {
                expect(message.description).toEqual('ResendRequest');
                expect(message.messageString).toMatchSnapshot();
                expect(fixServer.heartBeatInterval).toEqual(30);
                expect(fixServer.fixVersion).toEqual('FIX.4.6');
                fixParser.close();
                fixServer.destroy();
                done();
            });
            fixParser.on('error', (error) => {
                console.log('FIXParser: ', error);
            });
        });
    });
});
