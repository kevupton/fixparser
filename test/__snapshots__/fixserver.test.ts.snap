// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FIXServer TCP End-to-end: connect and Logon 1`] = `"8=FIX.4.79=7335=A34=149=CLIENT52=20210815-21:51:47.36556=SERVER141=Y98=0108=6410=210"`;

exports[`FIXServer TCP End-to-end: connect and Logon 2`] = `"8=FIX.4.79=7335=A34=149=CLIENT52=20210815-21:51:47.36556=SERVER141=Y98=0108=6410=210"`;

exports[`FIXServer TCP End-to-end: first message not a Logon 1`] = `"8=FIX.4.89=12935=D34=149=CLIENT52=20210815-21:51:47.36556=SERVER11=1122334421=138=12360=20210815-21:51:47.36540=154=155=700.HK59=010=095"`;

exports[`FIXServer TCP End-to-end: first message not a Logon 2`] = `"8=FIX.5.0SP29=8435=534=149=CLIENT52=20210815-21:51:47.36556=SERVER58=First message not a Logon10=019"`;

exports[`FIXServer TCP End-to-end: invalid Logon 1`] = `"8=FIX.5.09=8935=A34=149=INVALID_SENDER52=20210815-21:51:47.36556=INVALID_TARGET141=Y98=0108=6010=141"`;

exports[`FIXServer TCP End-to-end: invalid Logon 2`] = `"8=FIX.5.09=10635=534=149=INVALID_SENDER52=20210815-21:51:47.36556=INVALID_TARGET58=Invalid Logon TARGET or SENDER.10=230"`;

exports[`FIXServer TCP End-to-end: invalid MsgSeqNum 1`] = `"8=FIX.4.79=7535=A34=10049=CLIENT52=20210815-21:51:47.36556=SERVER141=Y98=0108=6410=052"`;

exports[`FIXServer TCP End-to-end: invalid MsgSeqNum 2`] = `"8=FIX.4.79=6435=234=149=CLIENT52=20210815-21:51:47.36556=SERVER7=116=010=241"`;

exports[`FIXServer Websocket End-to-end: connect and Logon 1`] = `"8=FIX.4.69=7635=A34=149=CLIENT252=20210815-21:51:47.36556=SERVER2141=Y98=0108=12810=105"`;

exports[`FIXServer Websocket End-to-end: invalid Logon 1`] = `"8=FIX.5.09=8935=A34=149=INVALID_SENDER52=20210815-21:51:47.36556=INVALID_TARGET141=Y98=0108=6010=141"`;

exports[`FIXServer Websocket End-to-end: invalid Logon 2`] = `"8=FIX.5.09=10635=534=149=INVALID_SENDER52=20210815-21:51:47.36556=INVALID_TARGET58=Invalid Logon TARGET or SENDER.10=230"`;

exports[`FIXServer Websocket End-to-end: invalid MsgSeqNum 1`] = `"8=FIX.4.69=7835=A34=20049=CLIENT252=20210815-21:51:47.36556=SERVER2141=Y98=0108=12810=204"`;
